pkg/event/event.go

```diff
@@ -130,3 +131,27 @@ func (e Event) String() string {
 	}
 	return b.String()
 }
+
+func (e Event) Clone() Event {
+	new := Event{}
+	new.Context = e.Context.Clone()
+	new.DataBinary = e.DataBinary
+	new.DataEncoded = e.DataEncoded
+	new.Data = e.cloneData()
+	new.FieldErrors = e.FieldErrors
```

Comment:
The map is a pointer too, you will have to deep copy it too

https://play.golang.org/p/ptXhUuQrdXl

Reply:
Damn I forgot it